### PR TITLE
Functional tests - Sort profiles

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Sort profile
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="functional/BO/14_advancedParameters/06_team/profiles/05_sortProfiles" URL_FO=yourShopURL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/19067)

### CODE REVIEW

File: tests/puppeteer/campaigns/functional/BO/14_advancedParameters/06_team/profiles/05_sortProfiles.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    this.pageObjects = await init();
+  });
+  after(async () => {
+    await helper.closeBrowser(browser);
+  });
+
+  // Login into BO
+  loginCommon.loginBO();
+
+  it('should go to "Advanced parameters>Team" page', async function () {
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/functional/BO/14_advancedParameters/06_team/profiles/05_sortProfiles.js]:
```suggestion
  it('should go to \'Advanced parameters>Team\' page', async function () {
```

don't use " please

**Review by SimonGrn — DISMISSED** [review]

File: tests/puppeteer/campaigns/functional/BO/14_advancedParameters/06_team/profiles/05_sortProfiles.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    await this.pageObjects.boBasePage.goToSubMenu(
+      this.pageObjects.boBasePage.advancedParametersLink,
+      this.pageObjects.boBasePage.teamLink,
+    );
+    await this.pageObjects.boBasePage.closeSfToolBar();
+    const pageTitle = await this.pageObjects.employeesPage.getPageTitle();
+    await expect(pageTitle).to.contains(this.pageObjects.employeesPage.pageTitle);
+  });
+
+  it('should go to "Profiles" page', async function () {
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/functional/BO/14_advancedParameters/06_team/profiles/05_sortProfiles.js]:
```suggestion
  it('should go to \'Profiles\' page', async function () {
```

File: tests/puppeteer/campaigns/functional/BO/14_advancedParameters/06_team/profiles/05_sortProfiles.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    await testContext.addContextItem(this, 'testIdentifier', 'resetFilterFirst', baseContext);
+    numberOfProfiles = await this.pageObjects.profilesPage.resetAndGetNumberOfLines();
+    await expect(numberOfProfiles).to.be.above(0);
+  });
+
+  const tests = [
+    {args: {testIdentifier: 'sortByIDDesc', sortBy: 'id_profile', sortDirection: 'desc'}},
+    {args: {testIdentifier: 'sortByIDAsc', sortBy: 'id_profile', sortDirection: 'asc'}},
+    {args: {testIdentifier: 'sortByNameDesc', sortBy: 'name', sortDirection: 'desc'}},
+    {args: {testIdentifier: 'sortByNameAsc', sortBy: 'name', sortDirection: 'asc'}},
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/functional/BO/14_advancedParameters/06_team/profiles/05_sortProfiles.js]:
```suggestion
    {args: {testIdentifier: 'sortByIDDesc', sortBy: 'id_profile', sortDirection: 'desc'}},
    {args: {testIdentifier: 'sortByNameDesc', sortBy: 'name', sortDirection: 'desc'}},
    {args: {testIdentifier: 'sortByNameAsc', sortBy: 'name', sortDirection: 'asc'}},
    {args: {testIdentifier: 'sortByIDAsc', sortBy: 'id_profile', sortDirection: 'asc'}},
```
should be back to default sort value at the end

File: tests/puppeteer/campaigns/functional/BO/14_advancedParameters/06_team/profiles/05_sortProfiles.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+// Importing pages
+const BOBasePage = require('@pages/BO/BObasePage');
+const LoginPage = require('@pages/BO/login');
+const DashboardPage = require('@pages/BO/dashboard');
+const EmployeesPage = require('@pages/BO/advancedParameters/team/index');
+const ProfilesPage = require('@pages/BO/advancedParameters/team/profiles/index');
+// Test context imports
+const testContext = require('@utils/testContext');
+
+const baseContext = 'functional_BO_advancedParams_team_profiles_sortProfiles';
```
Comment (Progi1984) [inline tests/puppeteer/campaigns/functional/BO/14_advancedParameters/06_team/profiles/05_sortProfiles.js]:
```suggestion
const baseContext = 'functional_BO_advancedParameters_team_profiles_sortProfiles';
```

File: tests/puppeteer/campaigns/functional/BO/14_advancedParameters/06_team/profiles/05_sortProfiles.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  loginCommon.loginBO();
+
+  it('should go to \'Advanced parameters>Team\' page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'goToAdvancedParamsPage', baseContext);
+    await this.pageObjects.boBasePage.goToSubMenu(
+      this.pageObjects.boBasePage.advancedParametersLink,
+      this.pageObjects.boBasePage.teamLink,
+    );
+    await this.pageObjects.boBasePage.closeSfToolBar();
+    const pageTitle = await this.pageObjects.employeesPage.getPageTitle();
```
Comment (Progi1984) [inline tests/puppeteer/campaigns/functional/BO/14_advancedParameters/06_team/profiles/05_sortProfiles.js:54]: employeesPage ?

Comment (nesrineabdmouleh) [reply tests/puppeteer/campaigns/functional/BO/14_advancedParameters/06_team/profiles/05_sortProfiles.js:54]: Yes, Team/Employees page.

File: tests/puppeteer/campaigns/functional/BO/14_advancedParameters/06_team/profiles/05_sortProfiles.js (hunk 4 above)
Comment (nesrineabdmouleh) [reply tests/puppeteer/campaigns/functional/BO/14_advancedParameters/06_team/profiles/05_sortProfiles.js]: Done thx :)

File: tests/puppeteer/campaigns/functional/BO/14_advancedParameters/06_team/profiles/05_sortProfiles.js (hunk 3 above)
Comment (nesrineabdmouleh) [reply tests/puppeteer/campaigns/functional/BO/14_advancedParameters/06_team/profiles/05_sortProfiles.js]: Done thx :)

**Review by NeOMakinG — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Comment by Progi1984** [comment]
Hello @nesrineabdmouleh, thanks for your contribution, would you mind to rebase your pull request? Thanks

**Review by NeOMakinG — APPROVED** [review]

**Review by SimonGrn — APPROVED** [review]

**Pull request merged**